cpu/rpx0xx/dist/pioasm.bash

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
```

Comment:
using  bash to generate a header... sorry why not a patch to pioasm to add a riot output or a small change to the sdk  format?

Reply:
Could be a good idea to maintain a patch that adds a `riot_output.cpp` to the pioasm.
This would bypass the problem of possible shell incompatibilities. Using python, we would also bypass the problem. But I am not a python lover, so I like your idea more.

Reply:
thx looks much better now. Is there a reason why:
the instructions and the configuration are not part of the pio_program_t struct?
If all would be one struct, pio_alloc_program, pio_write_program, and pio_sm_init_common could all be one function for the users world.

Reply:
The `pio_program_t` struct is a datatype of the public PIO API `drivers/include/periph/pio.h`
It sould be as general as possible, fast copiable, and serves allocation and deallocation of instruction memory.

The `pio_program_conf_t` is a CPU internal datatype because different CPUs supporting PIO, may offer different configuration options. Hence, a CPU in RIOT supporting PIO, should provide that datatype.
If we put the configuration struct in the program, then that abstration is gone.
We should not put a `pio_program_conf_t *` in the program struct either because multiple instances of a program can be written to the instruction memory. They would have different locations, but share the same configuration. Hence, a configuration pointer in the program struct would be an unnecessary replication.

The reason why there is no reference of instructions inside the program struct is similar.
The CPU defines what a `pio_instr_t` is. It may possibly also be a 32 bit integer.
And multiple instances of a program share the same instructions, so we do not want to unneseccarily replicate a pointer. The pico SDK creates a `static const uint16_t[]` of instructions. But we just `#define` the array content. The advantage of that is that we do not have to waste ROM for short programs that can be easily allocated within a function and written to the instruction memory.